src/core/stdcpp/vector.d

```diff
@@ -114,6 +114,18 @@ extern(D):
     ///
     void append(T[] array)                                                  { insert(length, array); }
 
+    /// Performs elementwise equality check.
+    bool opEquals(this This, That)(auto ref That rhs)
+    if (is(immutable That == immutable vector))                             { return as_array == rhs.as_array; }
+
+    /// Performs lexicographical comparison.
+    static if (is(typeof((ref T a, ref T b) => a < b)))
```

Comment:
The `static if` is so `__traits(hasMember, vector!T, "opCmp")` will be false for vectors that do not support comparison.